### ISSUE TITLE
Add support for fetching a subgraph by Translator.get_instance()

### ISSUE THREAD

**Issue by mdkozlowski** [memgraph/gqlalchemy#319]
`Translator._parse_memgraph` uses `get_edges_from_db`, which matches all edges in Memgraph. For large graphs this is slow and possibly unnecessary (like for my use-case).

I've rewritten the `Translator` and `DGLTranslator` classes for my use-case, by adding a Match parameter in `.get_instance` which is propagated down to the `get_edges_from_db` function. By providing a Match query on node labels and relationship properties, I can use the customised DGLTranslator to return subgraphs as part of a PyTorch/DGL Dataloader - making it very convenient for model training/prediction.

Is there a technical/conceptual reason the existing Translator class doesn't support optionally fetching a subgraph? My solution is specific to the use-case I have, and as-is, it definitely isn't ready for a PR. But if there is interest for this kind of functionality, I'd be happy to contribute.

**Comment by antejavor** [comment]
Hi @mdkozlowski, thanks for bringing this to our attention. 🚀 

There is no technical reason for it not to allow optionality.  It is more driven by time, effort, and priority. 

Since this is possible, we would gladly accept your contribution there. Once you have something more general, feel free to open a PR so we can form a platform for the whole community to benefit from this.

**Comment by katarinasupe** [comment]
Hi @mdkozlowski would you like to contribute this for the next version of GQLAlchemy, planned for March 12th? https://github.com/memgraph/gqlalchemy/milestone/7
